### DISCUSSION TITLE
Use Poetry for package management

### DISCUSSION THREAD

**Discussion by Aceticia** [ashleve/lightning-hydra-template#410] [Ideas]
With [this issue](https://github.com/python-poetry/poetry/issues/4231) looking like about to close, would it be a good idea to switch to poetry for dependency management & package installing? I'm still experimenting how well poetry works with torch and cuda, but if it does work I see it as a nice upgrade from `requirements.txt`.

This provides a few advantages:
1. All dependencies can be specified together with certain other python package options in `pyproject.toml`
2. Dev dependencies and normal dependencies can be more easily separated
3. Automatically generated lock files are more reliable for reproducibility compared to `requirements.txt`
4. Adding dependencies to an existing environment is easier

**Comment by cs-mshah** [comment]
I support this.
